README.md

```diff
@@ -1,2 +1,145 @@
 # tally-cpp
-C++ tally client
+
+Fast, buffered, hierarchical stats collection in C++.
+
+## Abstract
+
+Tally provides a common interface for emitting metrics, while letting you not worry about the
+velocity of metrics emission.
+
+By default it buffers counters, gauges and histograms at a specified interval, but does not
+buffer timer values.  This is primarily so timer values can have all their values sampled
+if desired and if not they can be sampled as summaries or histograms independently by a reporter.
+
+## Structure
+
+- **Scope**: Keeps track of metrics and their common metadata.
+- **Metrics**: Counters, Gauges, Timers and Histograms.
+- **Reporter**: Implemented by you. Forwards aggregated values from a scope to your metrics
+ingestion pipeline. A default implementation for emitting tagged metrics using Thrift is provided.
+
+### Usage
+
+```c++
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include <m3/reporter.h>
+#include <m3/reporter_builder.h>
+#include <tally/buckets.h>
+#include <tally/scope.h>
+#include <tally/scope_builder.h>
+
+int main(int argc, char** argv) {
+  // Create a reporter that will be used to emit your metrics.
+  auto reporter =
+      m3::ReporterBuilder()
+          .host("127.0.0.1")
```

Comment:
can we rely on defaults for these?

Reply:
These are the defaults actually, I just thought it would be clearer to specify them explicitly in the example.